app/services/filter_set.rb

```diff
@@ -0,0 +1,48 @@
+class FilterSet
```

Comment:
This is tested mostly at the feature level, I'm not sure unit tests would do a lot for us given the controller and feature specs that exist, and this mostly exercises rails model scopes.